Remove compilation warnings

During compilation, I get:

```
clang -D__KERNEL__ -D__ASM_SYSREG_H \
        -DCIRCLE_BUILD_URL=\"\" \
        -Wno-unused-value -Wno-pointer-sign -Wno-compare-distinct-pointer-types \
        -O2 -emit-llvm -c tcptracer-bpf.c \
         -I /usr/src/kernels/4.9.7-101.fc24.x86_64/arch/x86/include -I /usr/src/kernels/4.9.7-101.fc24.x86_64/arch/x86/include/generated -I /usr/src/kernels/4.9.7-101.fc24.x86_64/include -I /usr/src/kernels/4.9.7-101.fc24.x86_64/include/generated/uapi -I /usr/src/kernels/4.9.7-101.fc24.x86_64/arch/x86/include/uapi -I /usr/src/kernels/4.9.7-101.fc24.x86_64/include/uapi \
        -o - | llc -march=bpf -filetype=obj -o "/dist/tcptracer-ebpf.o"
In file included from tcptracer-bpf.c:3:
In file included from /usr/src/kernels/4.9.7-101.fc24.x86_64/include/linux/ptrace.h:5:
In file included from /usr/src/kernels/4.9.7-101.fc24.x86_64/include/linux/sched.h:61:
/usr/src/kernels/4.9.7-101.fc24.x86_64/include/linux/cgroup-defs.h:322:16: warning: field 'cgrp' with variable sized type 'struct cgroup' not at the end of a struct or class is a GNU extension [-Wgnu-variable-sized-type-not-at-end]
        struct cgroup cgrp;
                      ^
In file included from tcptracer-bpf.c:9:
In file included from /usr/src/kernels/4.9.7-101.fc24.x86_64/include/net/sock.h:51:
In file included from /usr/src/kernels/4.9.7-101.fc24.x86_64/include/linux/netdevice.h:38:
In file included from /usr/src/kernels/4.9.7-101.fc24.x86_64/include/linux/dmaengine.h:20:
In file included from /usr/src/kernels/4.9.7-101.fc24.x86_64/include/linux/device.h:24:
In file included from /usr/src/kernels/4.9.7-101.fc24.x86_64/include/linux/pinctrl/devinfo.h:21:
In file included from /usr/src/kernels/4.9.7-101.fc24.x86_64/include/linux/pinctrl/consumer.h:17:
In file included from /usr/src/kernels/4.9.7-101.fc24.x86_64/include/linux/seq_file.h:10:
/usr/src/kernels/4.9.7-101.fc24.x86_64/include/linux/fs.h:2693:9: warning: comparison of unsigned enum expression < 0 is always false [-Wtautological-compare]
        if (id < 0 || id >= READING_MAX_ID)
```

We should address the warning a compile with `-Werror`

```
/usr/src/kernels/4.9.7-101.fc24.x86_64/include/linux/fs.h:2693:9: warning: comparison of unsigned enum expression < 0 is always false
```

This warning seems legitimate BTW

--- a/tcptracer-bpf.c
+++ b/tcptracer-bpf.c
@@ -1,12 +1,18 @@
 #include <linux/kconfig.h>
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wgnu-variable-sized-type-not-at-end"
 #include <linux/ptrace.h>
+#pragma clang diagnostic pop
 #include <linux/version.h>
 #include <linux/bpf.h>
 #include "bpf_helpers.h"
 #include "tcptracer-bpf.h"
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wtautological-compare"
 #include <net/sock.h>
+#pragma clang diagnostic pop
 #include <net/inet_sock.h>
 #include <net/net_namespace.h>
 

